bitcoin/src/crypto/sighash.rs

```diff
@@ -515,6 +529,25 @@ impl TapSighashType {
             x => return Err(InvalidSighashTypeError(x as u32)), // Cast ok, u8 -> u32
         })
     }
+
+    /// Converts [`EcdsaSighashType`] to a `u8` sighash flag.
+    ///
+    /// The returned value is guaranteed to be a valid according to standardness rules.
+    pub fn to_u8(self) -> u8 { self as u8 }
+
+    /// Converts [`EcdsaSighashType`] to a `u32` sighash flag.
+    ///
+    /// The returned value is guaranteed to be a valid according to standardness rules.
+    pub fn to_u32(self) -> u32 { self as u32 }
```

Comment:
Taproot no longer has that `u8` vs `u32` weirdness and doesn't use `u32` at all.

Reply:
Yes, I added both functions for API uniformity since knowing to use `to_u8` vs `to_u32` requires remembering that it was fixed in taproot.

Reply:
That doesn't make sense. If you use `to_u32` in Taproot you'll get broken signature production/verification. Code uniformity doesn't trump correctness. Nowhere in Rust have I seen a broken method added just for uniformity

Reply:
Point taken, will remove.